<launch>
    <arg name="marker_size" default="14.4" />
    <arg name="target_id" default="5" />
    <arg name="max_new_marker_error" default="0.1" /> <!--0.08 -->
    <arg name="max_track_error" default="1" />  <!--0.2 -->
    <arg name="cam_image_topic" default="image_raw" />
    <arg name="cam_info_topic" default="camera_info" />	
    <arg name="output_frame" default="camera" />
    <arg name="frequency" default="20" />
    <arg name="namespace" default="f550" />

    <include file="$(find ar_marker_client)/launch/blackfly.launch">
        <arg name="namespace" value="$(arg namespace)"/>
    </include>

    <node name="ar_track_alvar" pkg="ar_track_alvar" respawn="false" output="screen" type="individualMarkersNoKinect" ns="$(arg namespace)" args="$(arg marker_size) $(arg max_new_marker_error) $(arg max_track_error) $(arg cam_image_topic) $(arg cam_info_topic) $(arg output_frame) $(arg frequency)"/>
    
    <include file="$(find ar_marker_client)/launch/multi_marker_tracker.launch">
        <arg name="namespace" value="$(arg namespace)"/>
    </include>

    <node name="optitrack_relative_to_marker" pkg="multirotor_transformations" type="marker_optitrack_node" ns="$(arg namespace)">
        <remap from="vrpn_client_node/uav/pose" to="/vrpn_client_node/f550/pose"/>
        <remap from="vrpn_client_node/marker/pose" to="/vrpn_client_node/marker/pose"/>
    </node>


</launch>
